mlir/include/mlir/Dialect/MIOpen/gridwise_gemm_params.h

```diff
@@ -617,11 +626,96 @@ class PopulateParams : public PopulateParamsBase {
     return 1;
   }
 
+  LogicalResult
+  CalculateBlockGemmPerformanceParameters(InitParamsNonXDL *param,
+                                          ConvolutionContext &ctx,
+                                          DerivedBlockGemmParams &derived) {
+
+    derived.gemmMLevel0Cluster = 0;
+    derived.gemmNLevel0Cluster = 0;
+    derived.gemmMLevel1Cluster = 0;
+    derived.gemmNLevel1Cluster = 0;
+
+    if (param->blockSize == 64) {
+      derived.gemmMLevel0Cluster = 4;
+      derived.gemmNLevel0Cluster = 4;
+      derived.gemmMLevel1Cluster = 2;
+      derived.gemmNLevel1Cluster = 2;
+    } else if (param->blockSize == 128) {
+      derived.gemmMLevel0Cluster = 4;
+      derived.gemmNLevel0Cluster = 4;
+      derived.gemmMLevel1Cluster = 4;
+      derived.gemmNLevel1Cluster = 2;
+    } else if (param->blockSize == 256) {
+      derived.gemmMLevel0Cluster = 4;
+      derived.gemmNLevel0Cluster = 4;
+      derived.gemmMLevel1Cluster = 4;
+      derived.gemmNLevel1Cluster = 4;
+    } else {
+      return failure();
+    }
+
+    if (!(param->gemmMPerThread >= 2 && param->gemmMPerThread <= 4))
+      return failure();
+
+    if (!(param->gemmNPerThread >= 2 && param->gemmNPerThread <= 4))
+      return failure();
+
+    if (!(param->gemmMPerBlock % param->gemmMPerThread == 0 &&
+          param->gemmNPerBlock % param->gemmNPerThread == 0))
+      return failure();
+
+    const auto threadGemmMPerBlock =
+        param->gemmMPerBlock / param->gemmMPerThread;
+    const auto threadGemmNPerBlock =
+        param->gemmNPerBlock / param->gemmNPerThread;
+
+    const auto threadGemmMPerCluster =
+        derived.gemmMLevel0Cluster * derived.gemmMLevel1Cluster;
+    const auto threadGemmNPerCluster =
+        derived.gemmNLevel0Cluster * derived.gemmNLevel1Cluster;
+
+    if (!(threadGemmMPerBlock % threadGemmMPerCluster == 0) &&
+        (threadGemmNPerBlock % threadGemmNPerCluster == 0))
+      return failure();
+
+    const auto clusterMPerBlock = threadGemmMPerBlock / threadGemmMPerCluster;
+    const auto clusterNPerBlock = threadGemmNPerBlock / threadGemmNPerCluster;
+
+    // inline asm only support clusterMPerBlock = 2 andclusterNPerBlock =
+    // 2
+    if (!(clusterMPerBlock == 2 && clusterNPerBlock == 2))
+      return failure();
+
+    return success();
+  }
+
+  void obtainGemmCWriteVecLen(ConvolutionContext &ctx, InitParamsNonXDL &params,
```

Comment:
I don't see this function used anywhere. Seems like a copy of `calculateGemmCDestDataPerWrite`.

Reply:
BTW, you used to macro the tuning parameter logic on GemmC to be off, [here]([url](https://github.com/whchung/llvm-project/pull/26#discussion_r444968168)). Did you make that change to fix anything? Does it have to be applied to the new GemmC logic as well to force the GemmCPerWrite to 1?